admin/src/js/controllers/multiple-user.js

```diff
@@ -0,0 +1,151 @@
+angular.module('controllers').controller('MultipleUserCtrl', function(
+  $scope,
+  $uibModalInstance,
+  $window,
+  CreateMultipleUser,
+  DB
+) {
+
+  'use strict';
+  'ngInject';
+
+  // $scope.hideFooter = true;
+  $scope.status = { uploading: false };
+  $scope.displayAddMultipleModal = true;
+  $scope.displayUnavailableModal = false;
+  $scope.displayUploadConfirm = false;
+  $scope.displayProcessingStatus = false;
+  $scope.displayFinishSummary = false;
+  $scope.outputFileUrl = '';
+  $scope.processTotal = '';
+  const USER_LOG_DOC_ID = 'bulk-user-upload-';
+
+  $scope.onCancel = function () {
+    $scope.clearScreen();
+    $uibModalInstance.dismiss();
+  };
+
+  const getLogsByType = (docPrefix) => {
+    return DB({ logsDB: true })
+      .allDocs({
+        startkey: docPrefix,
+        endkey: docPrefix + '\ufff0',
+        include_docs: true
+      })
+      .then(result => {
+        if (!result || !result.rows || !result.rows.length) {
+          return;
+        }
+        return result.rows
+          .map(row => row.doc)
+          .sort((a, b) => new Date(b.bulk_uploaded_on) - new Date(a.bulk_uploaded_on));
+      });
+  };
+
+  const prepareStringForCSV = (str) => {
+    if (!str) {
+      return str;
+    }
+    return '"' + str.replace(/"/g, '""') + '"';
+  };
+
+  const convertToCSV = doc => {
```

Comment:
I'd like to stay out of the CSV business and instead use a library. Then we don't have to worry about things like quoting/escaping.